components/camel-jms/src/main/java/org/apache/camel/component/jms/reply/MessageSelectorCreator.java

```diff
@@ -70,18 +72,18 @@ public synchronized String get() {
         return answer;
     }
 
-    public void onPut(String key) {
-        dirty = true;
-        correlationIds.add(key);
-    }
-
-    public void onRemove(String key) {
+    // Changes to live correlation-ids invalidate existing message selector
+    private void timeoutEvent(TimeoutMap.Listener.Type type, String cid) {
```

Comment:
Can we avoid the switch as its not common practice and maybe people dont know about how it works and would think that remove and put does nothing in this example etc. Make the code understandable and maintainable for everyone.

Reply:
To clarify, you want me remove the `Listener.Type` enum and add individual `onRemoved`, `onPut`, `onEvicted` methods?

It's a shame as, generally speaking, we lose the ability to specify the listener as a (private) method::reference and many users seem to always implement internal listener interfaces on their (public) class signature rather than creating an anon class to do the work behind the scenes. I guess those are the same users who will fail to pay attention to the `Type`. Is this really Camel's problem or theirs? Your call obviously :)

Reply:
No just rewrite the switch to use if .. elseif ... else style

Reply:
done, please check